File: SDKExamples/Chart/src/main/java/fragments/axes/DateTimeContinuousAxisFragment.java
package fragments.axes;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;

import com.telerik.manual.tests.R;
import com.telerik.widget.chart.engine.databinding.FieldNameDataPointBinding;
import com.telerik.widget.chart.visualization.cartesianChart.RadCartesianChartView;
import com.telerik.widget.chart.visualization.cartesianChart.axes.DateTimeContinuousAxis;
import com.telerik.widget.chart.visualization.cartesianChart.axes.LinearAxis;
import com.telerik.widget.chart.visualization.cartesianChart.series.categorical.BarSeries;
import com.telerik.widget.chart.visualization.cartesianChart.series.categorical.LineSeries;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.Random;

/**
 * Created by ginev on 11/20/2014.
 */
public class DateTimeContinuousAxisFragment extends Fragment {
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        FrameLayout rootView = (FrameLayout) inflater.inflate(R.layout.fragment_datetime_continuous_axis, container, false);
        rootView.addView(this.createChart());
        return rootView;
    }

    private RadCartesianChartView createChart(){
        RadCartesianChartView chart = new RadCartesianChartView(this.getActivity());

        BarSeries series = new BarSeries();

        DateTimeContinuousAxis axis = new DateTimeContinuousAxis();
        axis.setLabelFormat("MM/dd");
        series.setHorizontalAxis(axis);

        LinearAxis verticalAxis = new LinearAxis();
        verticalAxis.setLabelFormat("%.0f");
        series.setVerticalAxis(verticalAxis);

        series.setCategoryBinding(new FieldNameDataPointBinding("date"));
        series.setValueBinding(new FieldNameDataPointBinding("value"));

        series.setData(this.getData());

        chart.getSeries().add(series);

        return chart;
    }

    private ArrayList<DataEntity> getData(){
        Random numberGenerator = new Random();
        ArrayList<DataEntity> result = new ArrayList<DataEntity>(8);
        int startingMonth =0;
        for (int i = 0; i < 8; i++){

            DataEntity entity = new DataEntity();
            entity.value = numberGenerator.nextInt(10) + 1;
            Calendar date = Calendar.getInstance();
            date.set(Calendar.MONTH, startingMonth++);
            entity.date = date;
            if (i == 2 || i == 6){
                continue;
            }
            result.add(entity);
        }

        return result;
    }

    public class DataEntity{
        public Calendar date;
        public int value;
    }
}

File: Samples-Xamarin/Fragments/Chart/Axes/DateTimeContinuousAxisFragment.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Util;
using Android.Views;
using Android.Widget;
using Com.Telerik.Widget.Chart.Visualization.CartesianChart;
using Java.Util;
using Com.Telerik.Widget.Chart.Visualization.CartesianChart.Series.Categorical;
using Com.Telerik.Widget.Chart.Visualization.CartesianChart.Axes;
using Com.Telerik.Widget.Chart.Engine.Databinding;

namespace Samples
{
	public class DateTimeContinuousAxisFragment : Android.Support.V4.App.Fragment, ExampleFragment
	{
		public override View OnCreateView(LayoutInflater inflater, ViewGroup container,
			Bundle savedInstanceState) {
			// Inflate the layout for this fragment
			ViewGroup rootView = (ViewGroup)inflater.Inflate(Resource.Layout.fragment_chart_example, container, false);
			rootView.AddView(this.createChart());
			return rootView;
		}

		private RadCartesianChartView createChart(){
			RadCartesianChartView chart = new RadCartesianChartView(this.Activity);

			BarSeries series = new BarSeries();

			DateTimeContinuousAxis axis = new DateTimeContinuousAxis();
			axis.LabelFormat = "MM/dd";
			series.HorizontalAxis = axis;

			LinearAxis verticalAxis = new LinearAxis();
			verticalAxis.LabelFormat = "%.0f";
			series.VerticalAxis = verticalAxis;

			series.ValueBinding = new ValueBinding();
			series.CategoryBinding = new CategoryBinding();

			series.Data = this.getData();

			chart.Series.Add(series);

			return chart;
		}

		private ArrayList getData(){
			Java.Util.Random numberGenerator = new Java.Util.Random();
			ArrayList result = new ArrayList(8);
			int startingMonth =0;
			for (int i = 0; i < 8; i++){

				DataEntity entity = new DataEntity();
				entity.value = numberGenerator.NextInt(10) + 1;
				Calendar date = Calendar.Instance;
				date.Set(CalendarField.Month, startingMonth++);
				entity.date = date;
				if (i == 2 || i == 6){
					continue;
				}
				result.Add(entity);
			}

			return result;
		}

		public String Title() {
			return "Date Time Continuous";
		}


		public class CategoryBinding : DataPointBinding{
			public override Java.Lang.Object GetValue (Java.Lang.Object p0)
			{
				DataEntity entity = (DataEntity)p0;
				return entity.date;
			}

		}

		public class ValueBinding : DataPointBinding{
			public override Java.Lang.Object GetValue (Java.Lang.Object p0)
			{
				DataEntity entity = (DataEntity)p0;
				return entity.value;
			}

		}

		public class DataEntity : Java.Lang.Object{
			public Calendar date;
			public int value;
		}
	}
}

